### PR TITLE
check node/npm installation status on ubuntu before install it.

### PR DESCRIPTION
**Pull request by chenzaichun**
Ubuntu has some deps issues if node is installed with node source
repoistory. check command is exists before apt install.

This is a workaround for npm deps issue if it had been installed manually.

see #666

### CODE REVIEW

File: install-eaf.sh [hunk 1]
```diff
@@ -1,4 +1,4 @@
-#!/usr/bin/env sh
+#!/usr/bin/env bash
```
Comment (manateelazycat) [inline install-eaf.sh:1]: Can keep sh this place?

I remember some contributor change from bash to sh.

Comment (manateelazycat) [reply install-eaf.sh:1]: @chenzaichun Check this.

Comment (chenzaichun) [reply install-eaf.sh:1]: bash is installed on most of Linux distro and is the default shell. What shell should it support, ksh, csh, tcsh?

Comment (manateelazycat) [reply install-eaf.sh:1]: Check https://github.com/manateelazycat/emacs-application-framework/issues/631

Comment (chenzaichun) [reply install-eaf.sh:1]: Thanks. I'll update it later. by the way, As i know, $() is not support by all some shell, for example, csh is not support it.

Comment (chenzaichun) [reply install-eaf.sh]: I'll remove array usage and command -v calling like #631.

Comment (manateelazycat) [reply install-eaf.sh]: > I'll remove array usage and command -v calling like #631.

Nice ;)

**Pull request merged**